contrib/sinit/patches/sinit.patch

```diff
@@ -0,0 +1,31 @@
+Index: sinit/sinit/sinit.c
+===================================================================
+--- sinit.orig/sinit/sinit.c	2023-07-03 22:41:08.919523547 +0200
++++ sinit/sinit/sinit.c	2023-07-03 22:59:35.459486724 +0200
+@@ -6,6 +6,8 @@
+ #include <stdio.h>
+ #include <stdlib.h>
+ #include <unistd.h>
++#include <fcntl.h>
++#include <sys/ioctl.h>
+ 
+ #define LEN(x)	(sizeof (x) / sizeof *(x))
+ #define TIMEO	30
+@@ -32,7 +34,7 @@
+ int
+ main(void)
+ {
+-	int sig;
++	int sig, fd;
+ 	size_t i;
+ 
+ 	if (getpid() != 1)
+@@ -40,6 +42,8 @@
+ 	chdir("/");
+ 	sigfillset(&set);
+ 	sigprocmask(SIG_BLOCK, &set, NULL);
++	fd = open("/dev/tty", O_RDWR);
```

Comment:
AFAIU descriptors 0, 1, 2 are already opened and attached to `/dev/tty`, aren't they?

Reply:
Yes, that's right. I just added this `open` here, so that it is clear that we call `TIOCNOTTY` ioctl on the controlling terminal. But I guess, that we could simply call the ioctl on descriptor 0,1 or 2 (and maybe add an explaining comment).

Reply:
Please add a comment then.

Reply:
Added a comment in https://github.com/cahirwpz/mimiker/pull/1399/commits/506acc63e667c751409c6ffe20d37eeade9336cc